src/Avalonia.Controls/MenuItem.cs

```diff
@@ -306,6 +327,36 @@ protected override void OnDetachedFromLogicalTree(LogicalTreeAttachmentEventArgs
             }
         }
 
+        protected override void OnAttachedToVisualTree(VisualTreeAttachmentEventArgs e)
+        {
+            base.OnAttachedToVisualTree(e);
+
+            if (this.GetVisualParent() is IControl parent)
+            {
+                // HACK: This nasty but it's all WPF's fault. Grid uses an inherited attached
+                // property to store SharedSizeGroup state, except property inheritance is done
+                // down the logical tree. In this case, the control which is setting
```

Comment:
We could add inheritance policy option in property registration

Reply:
I think that would be complicating things too much. An inherited property here is simply the wrong way to do it IMO.